plugins/lookup/cluster_moid.py

```diff
@@ -0,0 +1,145 @@
+#!/usr/bin/python
+# -*- coding: utf-8 -*-
+# Copyright: (c) 2021, Alina Buzachis <@alinabuzachis>
+# GNU General Public License v3.0+ (see COPYING or https://www.gnu.org/licenses/gpl-3.0.txt)
+
+
+from __future__ import absolute_import, division, print_function
+
+__metaclass__ = type
+
+
+DOCUMENTATION = r'''
+lookup: cluster_moid
+short_description: Look up MoID for vSphere objects using vCenter REST API
+description:
+    - Returns Managed Object Reference (MoID) of the object contained in the specified path.
+options:
+    _terms:
+        description: paths to query.
+        required: true
+    vcenter_hostname:
+        description:
+            - The hostname or IP address of the vSphere vCenter.
+            - If the value is not specified in the task, the value of environment variable
+              C(VMWARE_HOST) will be used instead.
+        required: true
+        type: str
+    vcenter_password:
+        description:
+            - The vSphere vCenter password.
+            - If the value is not specified in the task, the value of environment variable
+              C(VMWARE_PASSWORD) will be used instead.
+        required: true
+        type: str
+    vcenter_rest_log_file:
+        description:
+            - You can use this optional parameter to set the location of a log file.
+            - This file will be used to record the HTTP REST interaction.
+            - The file will be stored on the host that run the module.
+            - If the value is not specified in the task, the value of environment variable
+              C(VMWARE_REST_LOG_FILE) will be used instead.
+        type: str
+    vcenter_username:
+        description:
+            - The vSphere vCenter username.
+            - If the value is not specified in the task, the value of environment variable
+              C(VMWARE_USER) will be used instead.
+        required: true
+        type: str
+    vcenter_validate_certs:
+        default: true
+        description:
+            - Allows connection when SSL certificates are not valid. Set to C(false) when
+              certificates are not trusted.
+            - If the value is not specified in the task, the value of environment variable
+              C(VMWARE_VALIDATE_CERTS) will be used instead.
+        type: bool
+author:
+    - Alina Buzachis <@alinabuzachis>
+version_added: 1.0.1
+requirements:
+    - vSphere 7.0.2 or greater
+    - python >= 3.6
+    - aiohttp
```

Comment:
Not really a blocker at this stage. This block is the same in all the plugin, we can rely on a documentation fragment instead. e.g: https://github.com/ansible-collections/community.vmware/blob/main/plugins/modules/vca_vapp.py#L101-L102
If we do so, we can also use it with the of modules.

Reply:
I tried to do that. Let me know please if that's fine.

Reply:
I tried to do that. Let me know if that's fine. Thanks.